code/src/components/ListSaved.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux';
+import { barcodeReducer } from 'reducers/barcodeReducer'
+import { PieChart } from "components/PieChart";
+import styled from 'styled-components';
+import { ButtonBracket } from 'lib/Buttons';
+
+
+export const ListSaved = () => {
+  const productsArray = useSelector(store => store.reducer.productsArray);
+  const dispatch = useDispatch();
+
+  return (
+    <Wrapper>
+      {productsArray.map((product, index) => (
+        <ChartContainer key={index}>
+          <h4>
+            {product.product_name}
+            <Remove type="button" onClick={() => dispatch(barcodeReducer.actions.removeProduct(index))}>
+              -
+            </Remove>
+          </h4>
+          <Click onClick={() => dispatch(barcodeReducer.actions.setProduct(product))}>
```

Comment:
Can this have a more descriptive name? Or can the onClick be set directly to the pie chart maybe?

Reply:
Sure!